sympy/sets/handlers/intersection.py

```diff
@@ -306,7 +306,7 @@ def intersection_sets(self, other): # noqa:F811
             # univarite imaginary part in same variable
             x, xis = zip(*[solve_linear(i, 0) for i in Mul.make_args(im) if n in i.free_symbols])
             if x and all(i == n for i in x):
-                base_set -= FiniteSet(xis)
+                base_set = FiniteSet(*xis)
```

Comment:
Should it be `base_set &= FiniteSet(*xis)`?

Reply:
If the imaginary part is never 0 then there is no portion of the base set that will ever be real so the empty set is returned. Otherwise wherever the imaginary portion of the base set is 0 -- *within the base set* -- is where the intersection will be real. So I think you are right, @oscarbenjamin . e.g. in the test that was changed, if the expression were changed to `(n - 1)*(n - S.Half)*I` then `xis` is `{1, 1/2}` but only 1 is in the base set of Integers, so the answer should be {1}.